src/crypto/slow-hash.c

```diff
@@ -198,6 +198,21 @@ extern int aesb_pseudo_round(const uint8_t *in, uint8_t *out, const uint8_t *exp
     } while (0)
 #endif
 
+#define VARIANT2_2_1() \
```

Comment:
Adding to what was said in comments: VARIANT2_2_1 seems to be excessive and useless here. It just XORs data into scratchpad, the result of this XOR is not used until we hit the same 64-byte line on some next iteration. It will be ~100 iterations later, at best, given 2 MB scratchpad size. The delay between store and use allows it to be done fully in background on ASIC while writing back to memory and in parallel with all other calculations. Only VARIANT2_2_2 does the job of breaking fused hardware implementation of "8-byte multiply then 8-byte add".